### DISCUSSION TITLE
Reuse image when switching pages with the same image in different sizes

### DISCUSSION THREAD

**Discussion by markspolakovs** [vercel/next.js#22191] [Ideas]
This may be easier to explain with an example: https://image-transition-repro-go7f6t673.vercel.app/ (source at https://image-transition-repro-go7f6t673.vercel.app/_src).

The index page ("page 1") has an image using `next/image` at 128x128px. "Page 2" has the same image but at 256x256px.

When navigating from page 1 to page 2, `next/image` generates a whole new optimised URL for the image, which means the browser cannot reuse the existing cached image, so there's a blank space until the "new" image can be loaded. (To make the effect more pronounced, enable "Disable cache" and "Throttling" in DevTools).

I've encountered this in the real world, where two pages have the same logo but at different sizes, causing a flash of missing content when switching pages.

I'd suggest that `next/image` could be made to be smart enough to realise that the two pages have the same image but in different sizes, so reuse the old image, asynchronously load the new size, and swap the two once the new one is ready and cached.

I'd be up for writing a PR to add this, but I've never contributed to Next before so might need some guidance :sweat_smile:

**Comment by nikita240** [comment]
It's crazy that this isn't implemented. Without this feature it's more like an "Image Unoptimization API".
